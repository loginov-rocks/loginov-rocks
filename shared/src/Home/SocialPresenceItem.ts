export interface SocialPresenceItem {
  title: string;
  url: string;
  url2?: string;
}
