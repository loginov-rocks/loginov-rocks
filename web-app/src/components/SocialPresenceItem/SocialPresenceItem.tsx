import * as React from 'react';

import { SOCIAL_PRESENCE_DRIVE2_TITLE } from 'Constants';
import { Link } from 'components/Link';

interface Props {
  secondaryUrl?: string;
  title: string;
  url: string;
}

export const SocialPresenceItem: React.FC<Props> = ({ secondaryUrl, title, url }) => (
  title === SOCIAL_PRESENCE_DRIVE2_TITLE
    ? (
      <>
        <Link href={url}>{title}</Link>
        {' '}
        &mdash; my Toyota Chaser GX71 blog, originally
        {' '}
        <Link href={secondaryUrl as string}>in Russian</Link>
      </>
    ) : (
      <Link href={url}>{title}</Link>
    )
);

SocialPresenceItem.defaultProps = {
  secondaryUrl: undefined,
};
