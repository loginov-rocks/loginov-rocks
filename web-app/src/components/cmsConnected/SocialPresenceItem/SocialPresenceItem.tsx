import * as React from 'react';

import { ExternalLink } from 'components/shared/ExternalLink';
import { SOCIAL_PRESENCE_DRIVE2_TITLE } from 'Constants';

interface Props {
  secondaryUrl?: string;
  title: string;
  url: string;
}

export const SocialPresenceItem: React.FC<Props> = ({ secondaryUrl, title, url }) => (
  title === SOCIAL_PRESENCE_DRIVE2_TITLE
    ? (
      <>
        <ExternalLink href={url}>{title}</ExternalLink>
        {' '}
        &mdash; my Toyota Chaser GX71 blog, originally
        {' '}
        <ExternalLink href={secondaryUrl as string}>in Russian</ExternalLink>
      </>
    ) : (
      <ExternalLink href={url}>{title}</ExternalLink>
    )
);

SocialPresenceItem.defaultProps = {
  secondaryUrl: undefined,
};
