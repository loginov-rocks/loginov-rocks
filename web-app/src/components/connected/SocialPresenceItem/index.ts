import { SocialPresenceItem } from './SocialPresenceItem';

export { SocialPresenceItem };
