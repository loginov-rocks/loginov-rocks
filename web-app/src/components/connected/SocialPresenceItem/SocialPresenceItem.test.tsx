/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { SocialPresenceItem } from './SocialPresenceItem';

it('matches snapshot', () => {
  const wrapper = shallow(<SocialPresenceItem title="LinkedIn" url="https://www.linkedin.com/in/loginov-rocks/" />);

  expect(wrapper).toMatchSnapshot();
});
