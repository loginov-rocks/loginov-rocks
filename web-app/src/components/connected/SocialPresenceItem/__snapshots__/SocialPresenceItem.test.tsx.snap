// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Link
  href="https://www.linkedin.com/in/loginov-rocks/"
>
  LinkedIn
</Link>
`;
